src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2012,10 +2065,22 @@
   <Node Name="BoundIndexOrRangePatternIndexerAccess" Base="BoundExpression">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow" />
 
-    <Field Name="Receiver" Type="BoundExpression" Null="disallow" />
-    <Field Name="LengthOrCountProperty" Type="PropertySymbol" Null="disallow" />
-    <Field Name="PatternSymbol" Type="Symbol" Null="disallow" />
+    <!-- An expression with type Index or Range -->
     <Field Name="Argument" Type="BoundExpression" Null="disallow" />
+
+    <Field Name="LengthOrCountAccess" Type="BoundExpression" SkipInVisitor="true" />
+
+    <!-- The receiver placeholder for length access -->
+    <Field Name="ReceiverPlaceholder" Type="BoundIndexOrRangeIndexerPatternReceiverPlaceholder" />
+
+    <!--
+    May be BoundIndexerAccess or BoundIndexOrRangePatternIndexerAccess,
+    which are built using the real receiver (not a placeholder) and placeholder(s) for argument(s).
+    -->
+    <Field Name="IndexerAccess" Type="BoundExpression" Null="disallow" SkipInVisitor="true" />
```

Comment:
>SkipInVisitor="true" [](http://example.com/codeflow?start=71&length=20)

The fact that we didn't need to do anything in SemanticModel bothers me somewhat. Are we still able to get semantic information for the receiver? GetTypeInfo/GetSymbolInfo and such?

Reply:
Adding test `SemanticModelOnReceiver` (looks good)

Reply:
> Adding test SemanticModelOnReceiver (looks good)

I debugged through the test. It looks to me that we definitely need to follow up on this. Since nothing appears to be obviously broken, a separate PR should be fine. Here is what is going wrong. As I suspected, we do not cache node for the receiver when we bind statement containing the element access. Therefore, we don't find a bound node for it on the first attempt. So we bind the containing statement again and still don't find the corresponding bound node in the map. Then we go into an error recovery mode and bind the receiver on its own. Besides utilizing the error recovery mode, we end up with a map that is inconsistent. For the containing element access we have a bound node that uses different node for the receiver. That is likely to create problems, if not today, then in the future.